packaging/conda/build.sh

```diff
@@ -1,12 +1,109 @@
-# !/bin/bash
+#!/bin/bash
 
-# On macOS, we have to build with the compiler outside conda. With conda's clang the following error occurs:
-# ld: unsupported tapi file type '!tapi-tbd' in YAML file '/Applications/Xcode_14.2.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/usr/lib/libSystem.tbd' for architecture x86_64
-if [ "$(uname)" == "Darwin" ]
+# Set-up the shell to behave more like a general-purpose programming language
+set -euo pipefail
+
+# Build MODL and MODL_Coclustering
+# Note on macOS we need the macOS SDK 10.10 for this conda build to work
+if [[ "$(uname)" == "Darwin" ]]
 then
-    cmake --preset macos-clang-release -DCMAKE_CXX_COMPILER=/usr/bin/clang++
-    cmake --build --preset macos-clang-release --parallel --target MODL MODL_Coclustering
+  CMAKE_PRESET="macos-clang-release"
 else
-    cmake --preset linux-gcc-release
-    cmake --build --preset linux-gcc-release --parallel --target MODL MODL_Coclustering
+  CMAKE_PRESET="linux-gcc-release"
+fi
+cmake --preset $CMAKE_PRESET -DBUILD_JARS=OFF -DTESTING=OFF -DCMAKE_CXX_COMPILER="$PREFIX/bin/mpicxx"
```

Comment:
POurquoi est-ce qu'il faut ajouter le compilateur mpi ?

Reply:
For some reason, CMake doesn't detect MPI in the conda environment (it is very frustrating to debug). Giving the compiler explicitly solves this.

Reply:
Maybe I should add a comment ?

Reply:
In a conda env `$PREFIX/bin` is always on the path. I removed the line to test it, it is a different problem from what it was a while ago. It is now related to clang and in mac only (it happens in my ARM mac as well).

Here is the run log:
https://github.com/KhiopsML/khiops/actions/runs/8537294734/job/23387558996

Reply:
I have the same pbm with clang on my Linux laptop...
To sum up, it fixes a clang issue on macOS, yes ?
NB: I have the same pbm on Linux with clang: cmake doesn't find MPI. But nothing works so far

Reply:
> I have the same pbm with clang on my Linux laptop... To sum up, it fixes a clang issue on macOS, yes ? NB: I have the same pbm on Linux with clang: cmake doesn't find MPI. But nothing works so far

Yes, it fixes a macOS issue. I'll add comment.

In your case the problem isn't fixed even when giving  `-DCMAKE_CXX_COMPILER=mpicxx` ?

Reply:
I added it.